### PR TITLE
ci(android): run native android tests

### LINKED ISSUE #22: [CI] Run unit tests for Android implementation on CI
We already have tests for the android implementation but they are not been run on CI for each PR

# Folders and tests
So we have:
- `lib/`: library implementation, the interace that clients use.
   - `tests/`: test for the flutter side of the library
- `android/`
   - [android/src/test](https://github.com/la-haus/iterable-flutter/tree/master/android/src/test/java/com/lahaus/iterable_flutter): these are the tests we want to run

# What we have tried
- Run flutter plugin tests, running `flutter test` at the root of the project and it runs [test/iterable_flutter_test.dart](https://github.com/la-haus/iterable-flutter/blob/master/test/iterable_flutter_test.dart) but not android ones.

- We have seen these docs [flutter/wiki/Plugin-Tests](https://github.com/flutter/flutter/wiki/Plugin-Tests), but I still don't figure out how to run them.

### PR DESCRIPTION
**Pull request by Santi92**
Fix https://github.com/la-haus/iterable-flutter/issues/22

### CODE REVIEW

**Review by cdmunoz — APPROVED** [review]
Awesome! 💪🏻 🍄 

Remember to add a description to the PR 🤓

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+concurrency:
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    working-directory: example/
+
+jobs:
+  build_android:
```
Comment (danielgomezrico) [inline .github/workflows/android.yml]: It would be better to put `unit_test` here instead of `build_android` since this job is doing that

**Review by danielgomezrico — APPROVED** [review]
🤓  Beside the small comment it looks good to me, we need to merge it to see if it works right? I dont see the task running in the checks

**Comment by danielgomezrico** [comment]
I tried some things and got no good results, for some reason the android side of the flutter plugin does not generate a build version on which we could run the tests 🤔

**Comment by danielgomezrico** [comment]
We are not doing an effort here to make it work so, any PRs are welcome 🤓

**Pull request closed without merging**